internal/config/config.go

```diff
@@ -111,7 +117,43 @@ func (c *Config) EnableClassDiagrams(value *bool) {
 	c.ClassDiagrams.Enabled = value
 }
 
+// SetMode sets the Mode field to the provided value.
+// If the value is nil, the field is not changed.
+func (c *Config) SetMode(mode *string) {
+	if mode == nil || *mode == "" {
+		// No value passed, do nothing
+		return
+	}
+
+	c.Mode = *mode
+}
+
+// validateMode validates and normalizes the mode field.
+func (c *Config) validateMode() error {
```

Comment:
A validation method should never under any circumstances modify the object being validated - it should simply return an error (if there is a problem) or nothing (if not).